monai/transforms/spatial/array.py

```diff
@@ -1035,16 +1066,13 @@ def __init__(
         The affine transformations are applied in rotate, shear, translate, scale order.
 
         Args:
-            rotate_params (float, list of floats): a rotation angle in radians,
-                a scalar for 2D image, a tuple of 3 floats for 3D. Defaults to no rotation.
-            shear_params (list of floats):
-                a tuple of 2 floats for 2D, a tuple of 6 floats for 3D. Defaults to no shearing.
-            translate_params (list of floats):
-                a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Translation is in pixel/voxel
-                relative to the center of the input image. Defaults to no translation.
-            scale_params (list of floats):
-                a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Defaults to no scaling.
-            spatial_size (list or tuple of int): output image spatial size.
+            rotate_params: a rotation angle in radians, a scalar for 2D image, a tuple of 3 floats for 3D.
+                Defaults to no rotation.
+            shear_params: a tuple of 2 floats for 2D, a tuple of 6 floats for 3D. Defaults to no shearing.
+            translate_params: a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Translation is in
+                pixel/voxel relative to the center of the input image. Defaults to no translation.
+            scale_params: a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Defaults to no scaling.
+            spatial_size: output image spatial size.
```

Comment:
If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.